crates/subspace-archiving/src/archiver.rs

```diff
@@ -833,10 +869,14 @@ impl Archiver {
         self.buffer
             .push_front(SegmentItem::ParentSegmentHeader(segment_header));
 
-        NewArchivedSegment {
-            segment_header,
-            pieces: pieces.to_shared(),
-            object_mapping,
-        }
+        (
+            NewArchivedSegment {
+                segment_header,
+                pieces: pieces.to_shared(),
+            },
+            object_mapping
+                .try_into()
+                .expect("Constructed with this size, which is never changed; qed"),
```

Comment:
NIT: I wouldn't convert it into array here, 256 elements on the stack is a lot, returning `Vec<PieceObjectMapping>` is perfectly fine here, especially since in later commit you'll not expose `PieceObjectMapping` anymore anyway, it just makes diff larger here.